tests/net/socket/tcp/testcase.yaml

```diff
@@ -1,7 +1,4 @@
 tests:
   test:
-    extra_configs:
```

Comment:
The idea here was to make sure that the test works with both "normal" setup (the same as the samples are run with) and with CONFIG_NET_TEST=y. When something is made to work with only non-default setup, there's always a suspicion that some black voodoo is needed to make it work *at all*.